### PR TITLE
Add stacktrace to job error log message

### PR DESCRIPTION
**Pull request by rodrigues**
Without this, it's harder to see why the job errored.

We can see the error, the job where the error happened, the args, but not where it happened.

### CODE REVIEW

File: test/toniq/job_process_test.exs [hunk 1]
```diff
@@ -25,20 +25,22 @@ defmodule Toniq.JobProcessTest do
     assert Toniq.JobProcess.run(job) == {:job_was_successful, job}
   end
 
-  test "a job that raises an error returns {:job_has_failed, job, error}" do
+  test "a job that raises an error returns {:job_has_failed, job, error, stack}" do
     job = %{worker: TestErrorWorker, arguments: [data: 10]}
-    assert Toniq.JobProcess.run(job) == {:job_has_failed, job, %RuntimeError{message: "fail"}}
+    result = Toniq.JobProcess.run(job)
+    assert {:job_has_failed, ^job, %RuntimeError{message: "fail"}, _} = result
+    assert {_, _, _, [{TestErrorWorker, :perform, 1, _}|_]} = result
```
Comment (joakimk) [inline test/toniq/job_process_test.exs]: Looks good. A couple of things though before I merge this.

I've written the other tests using "assert actual == expected", so I'd prefer if these where like that too. If for nothing else then for consistency.

Another reason to use actual == expected is that then ExUnit can give us better error messages.

When using `expected = actual` and relying on pattern matching you get:



When doing`actual == expected` you get:



Also, do you think we need to assert something about the stack trace? This test would pass even if it was "nil".

Comment (rodrigues) [reply test/toniq/job_process_test.exs]: good point, @joakimk, just ammended the commit, making the stack trace check more explicit 🙂

**Comment by joakimk** [comment]
Released as 1.1.0.
